crates/bevy_app/src/lib.rs

```diff
@@ -24,12 +24,12 @@ pub mod prelude {
     };
 }
 
-use bevy_ecs::schedule::StageLabel;
+use bevy_ecs::schedule::IntoStageLabel;
 
 /// The names of the default [`App`] stages.
 ///
 /// The relative [`Stages`](bevy_ecs::schedule::Stage) are added by [`App::add_default_stages`].
-#[derive(Debug, Hash, PartialEq, Eq, Clone, StageLabel)]
+#[derive(Debug, Hash, PartialEq, Eq, Clone, IntoStageLabel)]
```

Comment:
As part of these changes, we should get rid of all of the extra derives on these types.

My first impression is that we should have a blanket Debug impl, remove Hash, and keep PartialEq, Eq and Clone.

Reply:
I didn't want to break any user code unnecessarily, and I don't see how the extra derives are hurting anyone. What if we only remove them from private definitions?